chat-android/src/main/java/com/ably/chat/RoomOptions.kt

```diff
@@ -1,113 +1,187 @@
         /**
          * Supports all room options with default values
          */
-        public val default: RoomOptions = RoomOptions(
-            typing = TypingOptions(),
-            presence = PresenceOptions(),
-            reactions = RoomReactionsOptions(),
-            occupancy = OccupancyOptions(),
-        )
+        public val AllFeaturesEnabled: RoomOptions = buildRoomOptions {
+            typing()
+            presence()
+            reactions()
+            occupancy()
+        }
     }
 }
 
 /**
  * Represents the presence options for a chat room.
  */
-public data class PresenceOptions(
+public interface PresenceOptions {
     /**
      * Whether the underlying Realtime channel should use the presence enter mode, allowing entry into presence.
      * This property does not affect the presence lifecycle, and users must still call [Presence.enter]
      * in order to enter presence.
      * @defaultValue true
      */
-    val enter: Boolean = true,
+    public val enter: Boolean
 
     /**
      * Whether the underlying Realtime channel should use the presence subscribe mode, allowing subscription to presence.
      * This property does not affect the presence lifecycle, and users must still call [Presence.subscribe]
      * in order to subscribe to presence.
      * @defaultValue true
      */
-    val subscribe: Boolean = true,
-)
+    public val subscribe: Boolean
+}
 
 /**
  * Represents the typing options for a chat room.
  */
-public data class TypingOptions(
+public interface TypingOptions {
     /**
      * The timeout for typing events in milliseconds. If typing.start() is not called for this amount of time, a stop
      * typing event will be fired, resulting in the user being removed from the currently typing set.
      * @defaultValue 5000
      */
-    val timeoutMs: Long = 5000,
-)
+    public val timeoutMs: Long
+}
 
 /**
  * Represents the reactions options for a chat room.
  *
  * Note: This class is currently empty but allows for future extensions
  * while maintaining backward compatibility.
  */
-public class RoomReactionsOptions {
-    override fun equals(other: Any?): Boolean = other is RoomReactionsOptions
-    override fun hashCode(): Int = javaClass.hashCode()
-}
+public interface RoomReactionsOptions
 
 /**
  * Represents the occupancy options for a chat room.
  *
  * Note: This class is currently empty but allows for future extensions
  * while maintaining backward compatibility.
  */
-public class OccupancyOptions {
-    override fun equals(other: Any?): Boolean = other is OccupancyOptions
-    override fun hashCode(): Int = javaClass.hashCode()
+public interface OccupancyOptions
+
+@ChatDsl
+public class MutableRoomOptions : RoomOptions {
+    override var presence: MutablePresenceOptions? = null
+    override var typing: MutableTypingOptions? = null
+    override var reactions: MutableRoomReactionsOptions? = null
+    override var occupancy: MutableOccupancyOptions? = null
+}
```

Comment:
```suggestion
public class MutableRoomOptions : RoomOptions {
    override var presence: MutablePresenceOptions? = null
    override var typing: MutableTypingOptions? = null
    override var reactions: MutableRoomReactionsOptions? = null
    override var occupancy: MutableOccupancyOptions? = null
    
        override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is MutableRoomOptions) return false

        return gson.toJson(this) == gson.toJson(other)
    }
}
```

I believe there's excessive duplication of `RoomOptions`-related data classes at the moment. We are only using them for comparison purposes. By using the code above, we can eliminate the need for both current and future duplicate `Equatable` data classes.

Reply:
First of all it's not a good idea to rely on serialization to check fields are equal it adds unnecessary overhead, that can lead to errors, also implementing `equals` without `hashCode` is a bad practice. 

And lastly we should get rid of Gson library anyway :)  it's not very good for android, since it relies heavily on reflection and strong dependency on it is not very good. Android developers have alternatives that are more popular: Jackson, Moshi and kotlinx.serialization

Reply:
Okay, adding `hashcode` should be simple 
```
    override fun hashCode(): Int {
        return gson.toJson(this).hashCode()
    }
```
I thought `gson` is default library is in android, but in fact `org.json` is the default one.

Reply:
Hmm... `gson` is third party dependency and `org.json` serialization needs explicit mention of properties : (

Reply:
We should be able to update above class to data class, it should avoid class duplication but not sure if its recommended.
```
public data class MutableRoomOptions(
    override var presence: MutablePresenceOptions? = null,
    override var typing: MutableTypingOptions? = null,
    override var reactions: MutableRoomReactionsOptions? = null,
    override var occupancy: MutableOccupancyOptions? = null,
): RoomOptions
```

Reply:
Unfortunately this way we once again create public data class, that it's better to avoid. We can of course mention in the docs that sdk users shouldn't use constructor and copy method. We can mark them with `@InternalApi` annotation. But I think having data classes for this use-case, it's not that much additional code to maintain. Anyway  we always can refactor this later

Reply:
Yeah, I guess we are okay for now 👍